src/baseVisitor.ts

```diff
@@ -348,9 +348,8 @@ export class BaseVisitor<T> implements Required<RecursiveVisitors<T>> {
       cb(n.body, st)
     }
   }
-  Identifier(n: swc.Identifier, st: T, cb: Callback<T>) {
+  Identifier<S>(n: swc.Identifier | swc.BindingIdentifier, st: S, cb: Callback<S>) {
     if ('typeAnnotation' in n && n.typeAnnotation) {
```

Comment:
thanks to the union, typescript can discriminate the node to `BindingIdentifier` if `typeAnnotation` is within the node